### PR TITLE
APPT-407: EULA agreement mechanism

### PR DESCRIPTION
**Pull request by sambiramairelogic**
Creates a new document in index_data which holds the latest EULA version. 

Adds a new field to the UserProfile object to record the latest version of the EULA (if any) they have consented to.

Creates API routes for getting the EULA document and also updating this latest accepted version property on a user profile. 

Creates a EULA frontend which the user will be redirected to until they have consented.

### CODE REVIEW

**Comment by ste-banks** [comment]
Got conflicts to resolve but all looks good to me. Will approve after those have been sorted :)

File: src/api/Nhs.Appointments.Core/IUserService.cs [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ﻿namespace Nhs.Appointments.Core;
 
 public interface IUserService
-{    
+{
+    Task<User?> GetUserAsync(string userId);
```
Comment (vicr1) [inline src/api/Nhs.Appointments.Core/IUserService.cs:5]: Do we need to mark reference type as nullable

File: src/api/Nhs.Appointments.Core/UserService.cs [hunk 2]
```diff
@@ -5,6 +5,11 @@ namespace Nhs.Appointments.Core;
 
 public class UserService(IUserStore userStore, IRolesStore rolesStore, IMessageBus bus) : IUserService
 {
+    public Task<User?> GetUserAsync(string userId)
```
Comment (vicr1) [inline src/api/Nhs.Appointments.Core/UserService.cs:8]: Nullable reference type not required?

File: tests/Nhs.Appointments.Api.Integration/Scenarios/Eula/ConsentToEula.feature [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿Feature: Eula
+
+  Scenario: Consent to the latest EULA
+    Given the latest EULA is as follows
+      | VersionDate |
+      | Today       |
+    And the current user has agreed the EULA on the following date
+      | VersionDate |
+      | Yesterday   |
+    And the current user agrees to a EULA with the following date
```
Comment (vicr1) [inline tests/Nhs.Appointments.Api.Integration/Scenarios/Eula/ConsentToEula.feature:10]: This should be When as it is the Act part of the test

File: tests/Nhs.Appointments.Api.Integration/Scenarios/Eula/GetEula.feature [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿Feature: Eula
+
+  Scenario: Can get the latest EULA version
+    Given the latest EULA is as follows
+      | VersionDate |
+      | Today       |
+    And I request the latest EULA version
```
Comment (vicr1) [inline tests/Nhs.Appointments.Api.Integration/Scenarios/Eula/GetEula.feature:7]: Should be When

**Review by vicr1 — APPROVED** [review]

**Pull request merged**